templates/base.jinja2

```diff
@@ -0,0 +1,38 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="utf-8">
+    <title>Python Learning Journal</title>
+    <!--[if lt IE 9]>
+    <script src="http://html5shiv.googlecode.com/svn/trunk/html5.js"></script>
+    <![endif]-->
+    <link href="/static/style.css" rel="stylesheet" type="text/css">
+  </head>
+  <body>
+    <header>
+      <a href="{{ request.route_url('home') }}" id="header-link">
+          <h1>Python Learning Journal</h1>
+          <h2>Nicholas Draper</h2>
+      </a>
+      <aside id="user-controls">
+        <ul>
+        {% if not request.authenticated_userid %}
+          <li><a href="{{ request.route_url('login') }}">log in</a></li>
+        {% else %}
+          <li><a href="{{ request.route_url('logout') }}">log out</a></li>
+        {% endif %}
+        </ul>
+      </aside>
+    </header>
+    <main>
+      <section id="content">
+      {% block body %}
+      {% endblock %}
+      </section>
+    </main>
+    <footer>
+      <a id="code-fellows" href="http://codefellows.org">Code Fellows</a>
+      <a id="linkedin" href="https://github.com/ndraper2">Github</a>
+    </footer>
+  </body>
+</html>
```

Comment:
even in html files it's considered best to have a newline at the end of the file.